model/pipeline/src/main/proto/metrics.proto

```diff
@@ -52,38 +62,160 @@ message Annotation {
+      annotations: [{
+        key: "description",
+        value: "URN utilized to report user metric."
+      }]
+    }];
+
+    // Represents the latest seen double value. The timestamp is used to
+    // provide an "ordering" over multiple values to determine which is the
+    // latest.
+    USER_LATEST_DOUBLE = 5 [(monitoring_info_spec) = {
+      urn: "beam:metric:user:latest_double:v1",
+      type: "beam:metrics:latest_double:v1",
+      required_labels: ["PTRANSFORM", "NAMESPACE", "NAME"],
+      annotations: [{
+        key: "description",
+        value: "URN utilized to report user metric."
+      }]
+    }];
+
+    // Represents the largest set of integer values seen across bundles.
+    USER_TOP_N_INT64 = 6 [(monitoring_info_spec) = {
+      urn: "beam:metric:user:top_n_int64:v1",
+      type: "beam:metrics:top_n_int64:v1",
+      required_labels: ["PTRANSFORM", "NAMESPACE", "NAME"],
+      annotations: [{
+        key: "description",
+        value: "URN utilized to report user metric."
+      }]
+    }];
+
+    // Represents the largest set of double values seen across bundles.
+    USER_TOP_N_DOUBLE = 7 [(monitoring_info_spec) = {
+      urn: "beam:metric:user:top_n_double:v1",
+      type: "beam:metrics:top_n_double:v1",
+      required_labels: ["PTRANSFORM", "NAMESPACE", "NAME"],
+      annotations: [{
+        key: "description",
+        value: "URN utilized to report user metric."
+      }]
+    }];
+
+    // Represents the smallest set of integer values seen across bundles.
+    USER_BOTTOM_N_INT64 = 8 [(monitoring_info_spec) = {
+      urn: "beam:metric:user:bottom_n_int64:v1",
+      type: "beam:metrics:bottom_n_int64:v1",
+      required_labels: ["PTRANSFORM", "NAMESPACE", "NAME"],
+      annotations: [{
+        key: "description",
+        value: "URN utilized to report user metric."
+      }]
+    }];
+
+    // Represents the smallest set of double values seen across bundles.
+    USER_BOTTOM_N_DOUBLE = 9 [(monitoring_info_spec) = {
+      urn: "beam:metric:user:bottom_n_double:v1",
+      type: "beam:metrics:bottom_n_double:v1",
+      required_labels: ["PTRANSFORM", "NAMESPACE", "NAME"],
+      annotations: [{
+        key: "description",
+        value: "URN utilized to report user metric."
+      }]
+    }];
+
+    // General monitored state information which contains structured information
+    // which does not fit into a typical metric format. See MonitoringTableData
+    // for more details.
+    //
+    // TODO(BEAM-9617): Support monitored state.
+    //
+    // USER_MONITORING_TABLE = XX [(monitoring_info_spec) = {
+    //   urn: "beam:metric:user:v1",
+    //   type: "beam:metrics:monitoring_table:v1",
+    //   required_labels: ["PTRANSFORM", "NAMESPACE", "NAME"],
+    //   annotations: [{
+    //     key: "description",
+    //     value: "URN utilized to report user monitoring data."
+    //   }]
+    // }];
+
+    ELEMENT_COUNT = 10 [(monitoring_info_spec) = {
```

Comment:
Nit: Should we jump to 100 (or alternatively have the user types be 1xxx) just to have nice consecutive numbers are more types get added?

Reply:
Having the actual monitoring specs that are related grouped together is all that matters, the numbers shouldn't matter.